Add delatin support and tesselator option (#1372)
visgl/loaders.gl: modules/terrain/docs/api-reference/terrain-loader.md

| `terrain.meshMaxError`     | `number`        | `10`      | Mesh error in meters. The output mesh is in higher resolution (more vertices) if the error is smaller.                                        |
| `terrain.bounds`           | `array<number>` | `null`    | Bounds of the image to fit x,y coordinates into. In `[minX, minY, maxX, maxY]`. If not supplied, x and y are in pixels relative to the image. |
| `terrain.elevationDecoder` | `object`        | See below | See below                                                                                                                                     |

> | `terrain.tesselator` | `string`        | `auto` | Library used for mesh reconstruction. If `auto`, [MARTINI](https://github.com/mapbox/martini) will be used if the the image is a square and if it's dimension is the power of 2. Otherwise, [DELATIN](https://github.com/mapbox/delatin) will be used instead, which has no input limitations. Can also specify desired tesselator `martini` or `delatin` directly.                                                                                                                                      |

Reviewer: ```suggestion
| `terrain.tesselator` | `string`        | `auto` | Library used for mesh reconstruction. If `'auto'`, [MARTINI](https://github.com/mapbox/martini) will be used if the the image is a square where both height and width are powers of 2. Otherwise, [Delatin](https://github.com/mapbox/delatin) will be used instead, which has no input limitations. Can also specify `'martini'` or `'delatin'` explicitly.                                                                                                                                      |
```

  "offset": 0
}
```

### tesselator

The choices for tesselator are as follows:

`auto`:

> - Chooses Martini if the possible, otherwise use Delatin instead.

Reviewer: ```suggestion
- Chooses Martini if possible (if the image is a square where both height and width are powers of 2), otherwise uses Delatin instead, which has no input image limitations.
```

`martini`:

> - Only works on square 2^n+1 x 2^n+1 grids.

Reviewer: ```suggestion
- Uses the [Martini](https://github.com/mapbox/martini) algorithm for constructing a mesh.
- Only works on square 2^n+1 x 2^n+1 grids.
```

- Generates a hierarchy of meshes (pick arbitrary detail after a single run)
- Optimized for meshing speed rather than quality.

`delatin`:

> - Works on arbitrary raster grids.

Reviewer: ```suggestion
- Uses the [Delatin](https://github.com/mapbox/delatin) algorithm for constructing a mesh.
- Works on arbitrary raster grids.
```
